server/routes/questionsRoute.test.ts

```diff
@@ -0,0 +1,216 @@
+        expect(res.text).toContain('value="BOSS CHAIR" checked')
+        expect(res.text).toContain('value="DOG SEARCH" checked')
+      })
+  })
+
+  it('form is prefilled with report answers, multiple questions answered', () => {
+    reportWithDetails.type = 'ASSAULT'
+
+    reportWithDetails.questions = [
+      {
+        code: 'QID-000000061279',
+        question: 'WHAT WAS THE MAIN MANAGEMENT OUTCOME OF THE INCIDENT',
+        additionalInformation: null,
+        responses: [
+          {
+            response: 'POLICE REFERRAL',
+            responseDate: null,
+            additionalInformation: null,
+            recordedBy: 'USER1',
+            recordedAt: incidentDateAndTime,
+          },
+        ],
+      },
+      {
+        code: 'QID-000000061280',
+        question: 'IS ANY MEMBER OF STAFF FACING DISCIPLINARY CHARGES',
+        additionalInformation: null,
+        responses: [
+          {
+            response: 'NO',
+            responseDate: null,
+            additionalInformation: null,
+            recordedBy: 'USER1',
+            recordedAt: incidentDateAndTime,
+          },
+        ],
+      },
+      {
+        code: 'QID-000000061281',
+        question: 'IS THERE ANY MEDIA INTEREST IN THIS INCIDENT',
+        additionalInformation: null,
+        responses: [
+          {
+            response: 'NO',
+            responseDate: null,
+            additionalInformation: null,
+            recordedBy: 'USER1',
+            recordedAt: incidentDateAndTime,
+          },
+        ],
+      },
+      {
+        code: 'QID-000000061282',
+        question: 'HAS THE PRISON SERVICE PRESS OFFICE BEEN INFORMED',
+        additionalInformation: null,
+        responses: [
+          {
+            response: 'NO',
+            responseDate: null,
+            additionalInformation: null,
+            recordedBy: 'USER1',
+            recordedAt: incidentDateAndTime,
+          },
+        ],
+      },
+      {
+        code: 'QID-000000061283',
+        question: 'IS THE LOCATION OF THE INCDENT KNOWN',
+        additionalInformation: null,
+        responses: [
+          {
+            response: 'YES',
+            responseDate: null,
+            additionalInformation: null,
+            recordedBy: 'USER1',
+            recordedAt: incidentDateAndTime,
+          },
+        ],
+      },
+    ]
+
+    return agent
+      .get(reportQuestionsUrl)
+      .redirects(1)
+      .expect(200)
+      .expect(res => {
+        expect(res.text).not.toContain('There is a problem')
+        expect(res.text).toContain('name="61279" type="radio" value="POLICE REFERRAL" checked')
+        expect(res.text).toContain('name="61280" type="radio" value="NO" checked')
+        expect(res.text).toContain('name="61281" type="radio" value="NO" checked')
+        expect(res.text).toContain('name="61282" type="radio" value="NO" checked')
+        expect(res.text).toContain('name="61283" type="radio" value="YES" checked')
```

Comment:
all these assertions rely on the current implementation of GOV.UK components.

i have an idea how to get round that but it's not a small change so i will park the thought for now!

Reply:
Yeah this is starting to be integration-ey but I thought it was good enough for now. I am not sure how to test just the value-reading controller logic tbh. But I thought there was some value in the having a route test that implicitly test that as well.

In practice I don't imagine the govuk-component markup to change that much anytime soon.